common-types/src/main/java/io/carbynestack/common/result/Success.java

```diff
@@ -105,6 +106,29 @@ public boolean isSuccess() {
         return this;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param consumer the consumer of {@link Success#value()}
+     * @param reason   the failure reason in case of the consumer throwing
+     *                 a {@code Throwable}
+     * @return {@code this}
+     * @throws NullPointerException if the consumer is {@code null}
+     * @version JDK 8
+     * @see #map(Function)
+     * @since 0.1.0
+     */
+    @Override
+    public Result<S, F> tryPeek(AnyThrowingConsumer<? super S> consumer, F reason) {
+        requireNonNull(consumer);
+        try {
+            consumer.accept(this.value());
+            return this;
+        } catch (Throwable throwable) {
+            return new Failure<>(reason);
```

Comment:
All information about the exception that occurred is lost and there is no way to recover the message or cause.
While the `reason` of the error might be sufficient to provide error details for a simple `consumer` logic, logging or further handling must be done in complex consumers themselves.

How about something like making `reason` capable of holding the Throwable.

Reply:
From a technical point of view, it is already possible to save a `Throwable` in reason, but this special implementation of peek currently deals with the simplest case (and the only one we have used so far). As soon as the described (more complex) case becomes relevant, we can take a similar path as with `Result.of`.